client/state/form/reducer.js

```diff
@@ -21,8 +22,12 @@ reducers[ SET_FORM_PROPERTY ] = ( state, action ) => {
 	return Object.assign( {}, state, newObj );
 };
 
-reducers[ NEXT_STEP ] = ( state ) => {
-	return Object.assign( {}, state, { currentStep: state.currentStep + 1 } );
+reducers[ BACK_FROM_SUGGESTION ] = ( state ) => {
+	return [
+		setFormProperty( 'errors', {} ),
+		setFormProperty( 'acceptSuggestion', undefined ),
+		setFormProperty( 'pristine', false ),
+	].reduce( form, state );
```

Comment:
Sorry, meant to comment on this - I had to do a bit of digging - this approach seems sound and is pretty much what abramov recommends here - https://twitter.com/dan_abramov/status/656074974533459968

In the comments on that tweet, he also advises against several dispatch calls as non-performant.